### PR TITLE
Client code changes (publisher only) with new GAX helpers

### PR DESCRIPTION
**Pull request by jskeet**
See https://github.com/googleapis/gax-dotnet/pull/12 for GAX changes
(This change won't build until GAX change is committed and published, but this is the result - with similar code in Subscriber.)

### CODE REVIEW

**Comment by jskeet** [comment]
Right, this is now ready to review and merge if you're happy.

File: src/Google.Pubsub.V1/PublisherClient.cs [hunk 1]
```diff
@@ -187,10 +159,7 @@ public static class ServiceDefaults
             ServiceEndpointSettings serviceEndpointSettings = null)
         {
             serviceEndpointSettings = serviceEndpointSettings ?? GetDefaultServiceEndpointSettings();
-            Channel channel = new Channel(
-                serviceEndpointSettings.Host ?? ServiceDefaults.Host,
-                serviceEndpointSettings.Port ?? ServiceDefaults.Port,
-                credentials);
+            Channel channel = ClientHelper.CreateChannel(serviceEndpointSettings, ServiceDefaults.Host, ServiceDefaults.Port, credentials);
```
Comment (chrisdunelm) [inline src/Google.Pubsub.V1/PublisherClient.cs]: I would prefer removing the mutation to serviceEndpointSettings:
Channel channel = ClientHelper.CreateChannel(serviceEndpointSettings ?? GetDefaultServiceEndpointSettings(),
    ServiceDefaults.Host, ServiceDefault.Port, credentials);
(and in SubscriberClient)

File: src/Google.Pubsub.V1/PublisherClient.cs [hunk 2]
```diff
@@ -318,36 +287,17 @@ public sealed partial class PublisherClientImpl : PublisherClient
...
-                this._calcDeadline = () => null;
-            }
-            else
-            {
-                IClock clock = settings.Clock ?? SystemClock.Instance;
-                TimeSpan timeout = settings.Timeout.Value;
-                this._calcDeadline = () => clock.GetCurrentDateTimeUtc() + timeout;
-            }
+            settings = settings ?? PublisherSettings.GetDefault();
+            _clientHelper = new ClientHelper(settings);
```
Comment (chrisdunelm) [inline src/Google.Pubsub.V1/PublisherClient.cs]: Could make this a one-liner:
_clientHelper = new ClientHelper(settings ?? PublisherSettings.GetDefault());
(and in SubscriberClient)

Comment (chrisdunelm) [inline src/Google.Pubsub.V1/PublisherClient.cs]: And I prefer the non-mutableness of this change.

**Comment by chrisdunelm** [comment]
LGTM
Looks even better to me, with from the minor changes proposed ;)

File: src/Google.Pubsub.V1/PublisherClient.cs (hunk 2 above)
Comment (jskeet) [inline src/Google.Pubsub.V1/PublisherClient.cs]: Yes, we can - so long as we don't need the settings for anything else. I strongly suspect that with bundling etc, we'll want to use the "effective settings" elsewhere in the constructor.

**Comment by jskeet** [comment]
Have one final look - will squash and merge if you're happy

**Comment by chrisdunelm** [comment]
LVGTM

**Pull request merged**